src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts

```diff
@@ -28,4 +28,5 @@ export class ProfileInfo {
     public accountName?: string;
     public versionControlUrl?: string;
     public programmingLanguageFeatures: ProgrammingLanguageFeature[];
+    public saml2?: { ['button-label']: string; ['enable-password']: boolean };
```

Comment:
Please extract two attributes here and avoid the use of { ... } as type

Reply:
Can this be achieved without changing the properties on the server side? The server does currently sends an object
```json
{
"saml2": { "button-label": "...", "enable-password": false }
}
```
Bundling related information would be preferable in my opinion.

Reply:
What you can do is create a new type for example in `login.service.ts` and export it:

```ts
export type Saml2 = { ['button-label']: string; ['enable-password']: boolean };
```

and then you can use that type instead of writing out the properties:

```ts
public saml2?: Saml2;
```

Reply:
We can convert what the server sends. This is more about the use of actual classes. So instead of using { ['button-label']: string; ['enable-password']: boolean } I propose to use something like

```
export class Saml2Config {
    public string buttonLabel;
    public boolean enablePassowrd;
    constructor(...) {}
}
```

Normally Angular http client can automatically transfer the json into such an object

Reply:
@ivanchimeno fixed this